src/python/gudhi/wasserstein/barycenter.py

```diff
@@ -33,37 +32,31 @@ def _mean(x, m):
 
 def lagrangian_barycenter(pdiagset, init=None, verbose=False):
     '''
-    :param pdiagset: a list of ``numpy.array`` of shape `(n x 2)`
-                    (`n` can variate), encoding a set of
-                    persistence diagrams with only finite coordinates.
+    :param pdiagset: a list of ``numpy.array`` of shape `(n x 2)` (`n` can variate), encoding a set of persistence
+        diagrams with only finite coordinates.
     :param init: The initial value for barycenter estimate.
-                    If ``None``, init is made on a random diagram from the dataset.
-                    Otherwise, it can be an ``int``
-                    (then initialization is made on ``pdiagset[init]``)
-                    or a `(n x 2)` ``numpy.array`` enconding
-                    a persistence diagram with `n` points.
+        If ``None``, init is made on a random diagram from the dataset.
+        Otherwise, it can be an ``int`` (then initialization is made on ``pdiagset[init]``)
+        or a `(n x 2)` ``numpy.array`` enconding a persistence diagram with `n` points.
     :type init: ``int``, or (n x 2) ``np.array``
-    :param verbose: if ``True``, returns additional information about the
-                    barycenter.
+    :param verbose: if ``True``, returns additional information about the barycenter.
     :type verbose: boolean
-    :returns: If not verbose (default), a ``numpy.array`` encoding
-              the barycenter estimate of pdiagset
-              (local minimum of the energy function).
-              If ``pdiagset`` is empty, returns ``None``.
-              If verbose, returns a couple ``(Y, log)``
-              where ``Y`` is the barycenter estimate,
-              and ``log`` is a ``dict`` that contains additional informations:
-
-              - `"groupings"`, a list of list of pairs ``(i,j)``.
-              Namely, ``G[k] = [...(i, j)...]``, where ``(i,j)`` indicates
-              that ``pdiagset[k][i]`` is matched to ``Y[j]``
-              if ``i = -1`` or ``j = -1``, it means they
-              represent the diagonal.
-
-              - `"energy"`, ``float`` representing the Frechet energy value obtained.
-              It is the mean of squared distances of observations to the output.
-
-              - `"nb_iter"`, ``int`` number of iterations performed before convergence of the algorithm.
+    :returns: If not verbose (default), a ``numpy.array`` encoding the barycenter estimate of pdiagset
+        (local minimum of the energy function).
+        If ``pdiagset`` is empty, returns ``None``.
+        If verbose, returns a couple ``(Y, log)`` where ``Y`` is the barycenter estimate,
+        and ``log`` is a ``dict`` that contains additional informations:
+
+        - `"groupings"`, a list of list of pairs ``(i,j)``.
+
```

Comment:
I think it would make sense to make the description of "groupings" a single paragraph (same for "energy"), but the choice it up to @tlacombe .

Reply:
After few tests, I think that if we don't mind having >80 char lines, the rendering is slightly better on the doc (.html) by putting everything on a same line (but we must keep blank lines in between).

However, it makes the output of `help(lagrangian_barycenter)` a bit harder to read (as it breaks the 80 char rule).

I have no opinion on which one is better ; I guess people are more likely to read the online doc rather than calling for `help(lagrangian_barycenter)`. I will suggest a change this way.